backend/src/main/controller/PromotionController.ts

```diff
@@ -39,14 +43,24 @@ export class PromotionController {
       const promotions = await getCustomRepository(
         PromotionRepository
       ).getAllPromotions(promotionQuery);
+
+      for (const promotion of promotions) {
+        const currPlaceID = promotion.placeId;
+        const locationDetails: CachingObject = await this.cachingService.getLatLonValue(
+          currPlaceID
+        );
+        promotion.lat = locationDetails.lat;
+        promotion.lon = locationDetails.lon;
+      }
```

Comment:
What do you think about using `Promise.all`, mainly for this performance reason https://stackoverflow.com/a/46086037.

I would maybe even go one step further and pass in the promotion so that the `cachingService` would set the lat lon for the promotion, so then it could be something like

```suggestion
      const promises = promotions.map(promotion => this.cachingService.setLatLon(promotion));
      // then return all promotions with their lat lon values
```

And you can also see how much more is abstracted away since we don't really want to show too many implementation details inside the controller.

Let me know your thoughts on this!

Reply:
Hmm yeah, it does look cleaner! I also didn't know that we shouldn't show too many details inside the Controller- why is that?

Reply:
>I also didn't know that we shouldn't show too many details inside the Controller- why is that?

Great question! I have very little experience, but I think cause generally, the controllers shouldn't really show too many implementation details. Right now we should show the minimum and delegate responsibility to our services. https://softwareengineering.stackexchange.com/questions/26438/how-much-business-logic-should-be-allowed-to-exist-in-the-controller-layer @andrewzulaybar maybe you can chime in on your thoughts for this too?

>I'm also unsure how to associate a particular promotion with their Caching Object with Promise.all?

Hmm I'm not sure what you mean.
I was thinking `CachingService` would then define it as
```
fun setLatLon(promotion: Promotion): Promise<Promotion> {
        const locationDetails: CachingObject = await this.cachingService.getLatLonValue(
          promotion.placeId
        );
        promotion.lat = locationDetails.lat;
        promotion.lon = locationDetails.lon;
        return promotion;
}
```

Reply:
Ooooh I see, thanks for the link!! And that makes sense, also wondering if that would still be considered Promise.all since we're using the JS map functionality?

Reply:
Hmm like it would return an array of promises right and then all you would do is `await Promise.all(promsies)`. 

Now that I think about it, do you think we could abstract it even more and be like
```suggestion
      await this.cachingService.setLatLonForPromotions(promotions)
```